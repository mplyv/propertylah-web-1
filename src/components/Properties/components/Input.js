import styles from '../property.module.css'

const Input = (props)=>{
    return(
        <div className={styles.horizontalContainerPopup}>
            <input id={props.options[0]} className={styles.input} type='number' placeholder='Min' value={props.min} onChange={props.onChange}></input>
            <input id={props.options[1]} className={styles.input} type='number' placeholder='Max' value={props.max} onChange={props.onChange}></input>
        </div>
    )
}

export default Input