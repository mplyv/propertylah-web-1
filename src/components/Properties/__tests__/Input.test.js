import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer'
import Input from '../components/Input';

//* cleanup after each test
afterEach(() => {
  cleanup();
});

let priceOptions = ['MinPrice','MaxPrice'];
let floorsizeOptions = ['MinFloorSize','MaxFloorSize'];
let PSFOptions = ['MinPSF','MaxPSF'];
let TOPYearOptions = ['MinTOP','MaxTOP']
let minPrice = 0;
let maxPrice = 100;
let minPSF = 0;
let maxPSF = 100;
let minFloorsize = 0;
let maxFloorsize = 100;
let minYear = 1990;
let maxYear = 2020;

test('Input.js should render 2 inputs with id given by options',()=>{
  render(<Input options={priceOptions} min={minPrice} max={maxPrice} onChange={jest.fn()}/>);
  const inputMin = screen.getByTestId(`test-${priceOptions[0]}`);
  const inputMax = screen.getByTestId(`test-${priceOptions[1]}`);
  expect(inputMin).toBeInTheDocument();
  expect(inputMax).toBeInTheDocument();
  expect(inputMin.id).toBe(priceOptions[0]);
  expect(inputMax.id).toBe(priceOptions[1]);
})

test('Input.js should render with min and max values given by min and max',()=>{
  render(<Input options={priceOptions} min={minPrice} max={maxPrice} onChange={jest.fn()}/>);
  const inputMin = screen.getByTestId(`test-${priceOptions[0]}`);
  const inputMax = screen.getByTestId(`test-${priceOptions[1]}`);
  expect(inputMin.value).toBe(minPrice.toString());
  expect(inputMax.value).toBe(maxPrice.toString());
})

test('match price snapshot',()=>{
  const tree = renderer.create(<Input options={priceOptions} min={minPrice} max={maxPrice} onChange={jest.fn()}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

test('match floorsize snapshot',()=>{
  const tree = renderer.create(<Input options={floorsizeOptions} min={minFloorsize} max={maxFloorsize} onChange={jest.fn()}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

test('match PSF snapshot',()=>{
  const tree = renderer.create(<Input options={PSFOptions} min={minPSF} max={maxPSF} onChange={jest.fn()}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

test('match TOPYear snapshot',()=>{
  const tree = renderer.create(<Input options={TOPYearOptions} min={minYear} max={maxYear} onChange={jest.fn()}/>).toJSON();
  expect(tree).toMatchSnapshot();
})