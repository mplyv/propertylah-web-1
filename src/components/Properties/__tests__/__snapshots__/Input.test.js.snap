// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match PSF snapshot 1`] = `
<div
  className="horizontalContainerPopup"
>
  <input
    className="input"
    data-testid="test-MinPSF"
    id="MinPSF"
    onChange={[MockFunction]}
    placeholder="Min"
    type="number"
    value={0}
  />
  <input
    className="input"
    data-testid="test-MaxPSF"
    id="MaxPSF"
    onChange={[MockFunction]}
    placeholder="Max"
    type="number"
    value={100}
  />
</div>
`;

exports[`match TOPYear snapshot 1`] = `
<div
  className="horizontalContainerPopup"
>
  <input
    className="input"
    data-testid="test-MinTOP"
    id="MinTOP"
    onChange={[MockFunction]}
    placeholder="Min"
    type="number"
    value={1990}
  />
  <input
    className="input"
    data-testid="test-MaxTOP"
    id="MaxTOP"
    onChange={[MockFunction]}
    placeholder="Max"
    type="number"
    value={2020}
  />
</div>
`;

exports[`match floorsize snapshot 1`] = `
<div
  className="horizontalContainerPopup"
>
  <input
    className="input"
    data-testid="test-MinFloorSize"
    id="MinFloorSize"
    onChange={[MockFunction]}
    placeholder="Min"
    type="number"
    value={0}
  />
  <input
    className="input"
    data-testid="test-MaxFloorSize"
    id="MaxFloorSize"
    onChange={[MockFunction]}
    placeholder="Max"
    type="number"
    value={100}
  />
</div>
`;

exports[`match price snapshot 1`] = `
<div
  className="horizontalContainerPopup"
>
  <input
    className="input"
    data-testid="test-MinPrice"
    id="MinPrice"
    onChange={[MockFunction]}
    placeholder="Min"
    type="number"
    value={0}
  />
  <input
    className="input"
    data-testid="test-MaxPrice"
    id="MaxPrice"
    onChange={[MockFunction]}
    placeholder="Max"
    type="number"
    value={100}
  />
</div>
`;
